src/coordinates/uniform_cartesian.hpp

```diff
@@ -58,98 +58,77 @@ class UniformCartesian {
     cell_volume_ = dx_[0] * dx_[1] * dx_[2];
   }
 
-  template <class... Args>
-  KOKKOS_FORCEINLINE_FUNCTION Real Volume(Args... args) const {
-    return cell_volume_;
-  }
-
-  template <class... Args>
-  KOKKOS_FORCEINLINE_FUNCTION Real Dx(const int dir, Args... args) const {
+  template <int dir, class... Args>
+  KOKKOS_FORCEINLINE_FUNCTION Real Dxc(Args... args) const {
     assert(dir > 0 && dir < 4);
     return dx_[dir - 1];
   }
 
-  template <class... Args>
-  KOKKOS_FORCEINLINE_FUNCTION Real Area(const int dir, Args... args) const {
+  template <int dir, int face, class... Args>
+  KOKKOS_FORCEINLINE_FUNCTION Real Dxf(Args... args) const {
+    assert(dir > 0 && dir < 4 && face > 0 && face < 4);
+    return dx_[face - 1];
+  }
```

Comment:
I'm not sure why... but this overload in particular seemed to throw C++'s type deduction through a loop. A number of places, where I had
```C++
const auto &coords = pmb->coords;
```
I had to change to
```C++
const parthenon::Coordinates_t &coords = pmb->coords;
```
for template resolution of, e.g.,
```C++
coords.DXf<1,1>(i)
```
to resolve properly.

Reply:
Hmm, is this `class... Args` causing the type deduction causing issues? Did it happen on multiple compilers?

Reply:
I'm not sure... that's the likeliest culprit but I didn't see what was wrong. Only tried on gnu.

Reply:
I don't think this is urgent to fix, by the way.

Reply:
This issue came up when I switched from function argument functions to template arguments in the Poisson example. 
```
/vast/home/forrestglines/code/parthenon-project/parthenon/example/poisson/poisson_package.cpp(233): error: expected an expression

/vast/home/forrestglines/code/parthenon-project/parthenon/example/poisson/poisson_package.cpp(162): error: expected an expression

2 errors detected in the compilation of "/vast/home/forrestglines/code/parthenon-project/parthenon/example/poisson/poisson_package.cpp".
```

Changing out `auto` fixed the compilation issue